src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1158,20 +1161,26 @@ public static bool IsWordChar(char ch)
             }
 
             // For non-ASCII, fall back to checking the Unicode category.
-            switch (CharUnicodeInfo.GetUnicodeCategory(ch))
-            {
-                case UnicodeCategory.UppercaseLetter:
-                case UnicodeCategory.LowercaseLetter:
-                case UnicodeCategory.TitlecaseLetter:
-                case UnicodeCategory.ModifierLetter:
-                case UnicodeCategory.OtherLetter:
-                case UnicodeCategory.NonSpacingMark:
-                case UnicodeCategory.DecimalDigitNumber:
-                case UnicodeCategory.ConnectorPunctuation:
-                    return true;
+            UnicodeCategory category = CharUnicodeInfo.GetUnicodeCategory(ch);
 
-                default:
-                    return false;
+#if NET7_0_OR_GREATER
+            if (Vector128.IsHardwareAccelerated)
+            {
+                return Vector128.EqualsAny(
+                    Vector128.Create((byte)category),
+                    Vector128.Create(((ulong)UnicodeCategory.UppercaseLetter << 56) | ((ulong)UnicodeCategory.LowercaseLetter << 48) |
+                                     ((ulong)UnicodeCategory.TitlecaseLetter << 40) | ((ulong)UnicodeCategory.ModifierLetter << 32) |
+                                     ((ulong)UnicodeCategory.OtherLetter << 24) | ((ulong)UnicodeCategory.NonSpacingMark << 16) |
+                                     ((ulong)UnicodeCategory.DecimalDigitNumber << 8) | ((ulong)UnicodeCategory.ConnectorPunctuation)).AsByte());
+            }
+            else
+#endif
+            {
+                return category is
+                    UnicodeCategory.UppercaseLetter or UnicodeCategory.LowercaseLetter or
+                    UnicodeCategory.TitlecaseLetter or UnicodeCategory.ModifierLetter or
+                    UnicodeCategory.OtherLetter or UnicodeCategory.NonSpacingMark or
+                    UnicodeCategory.DecimalDigitNumber or UnicodeCategory.ConnectorPunctuation;
```

Comment:
```suggestion
            const int Mask =
                1 << (int)UnicodeCategory.UppercaseLetter |
                1 << (int)UnicodeCategory.LowercaseLetter |
                1 << (int)UnicodeCategory.TitlecaseLetter |
                1 << (int)UnicodeCategory.ModifierLetter |
                1 << (int)UnicodeCategory.OtherLetter |
                1 << (int)UnicodeCategory.NonSpacingMark |
                1 << (int)UnicodeCategory.DecimalDigitNumber |
                1 << (int)UnicodeCategory.ConnectorPunctuation;
            
            return (Mask & (1 << (int)category)) != 0;
```

Is using the vector faster than a simple mask?

Reply:
Hah! Probably not.  Let me check.

Reply:
haha, I was about to type the same 😆:
```csharp
static bool Test(char ch, UnicodeCategory category)
{
    if (category <= UnicodeCategory.ConnectorPunctuation)
    {
        uint mask =
            (1 << (int)UnicodeCategory.LowercaseLetter) |
            (1 << (int)UnicodeCategory.TitlecaseLetter) |
            (1 << (int)UnicodeCategory.ModifierLetter) |
            (1 << (int)UnicodeCategory.OtherLetter) |
            (1 << (int)UnicodeCategory.NonSpacingMark) |
            (1 << (int)UnicodeCategory.DecimalDigitNumber) |
            (1 << (int)UnicodeCategory.ConnectorPunctuation);
        return ((mask >> ch) & 1) == 1;
    }
    return false;
}
```
```asm
; Method Prog:Test(ushort,int):bool
G_M42129_IG01:              ;; offset=0000H
G_M42129_IG02:              ;; offset=0000H
       83FA12               cmp      edx, 18
       7F11                 jg       SHORT G_M42129_IG05
G_M42129_IG03:              ;; offset=0005H
       0FB7C1               movzx    rax, cx
       BA3E010400           mov      edx, 0x4013E
       C4E2FBF7C2           shrx     rax, rdx, rax
       83E001               and      eax, 1
G_M42129_IG04:              ;; offset=0015H
       C3                   ret      
G_M42129_IG05:              ;; offset=0016H
       33C0                 xor      eax, eax

G_M42129_IG06:              ;; offset=0018H
       C3                   ret      
; Total bytes of code: 25
```
Technically, JIT emits something similiar for SWITCH opcodes, but Roslyn doesn't want to emit a switch here

Reply:
ah, yeah, condition `if (category <= UnicodeCategory.ConnectorPunctuation)` is not needed here.

Still, the codegen can be better if we use `bt` instruction here